extendr-api/src/robj/mod.rs

```diff
@@ -734,65 +733,143 @@ where
 {
     fn as_typed_slice(&self) -> Option<&'a [T]>
     where
-        Self: 'a,
-    {
-        None
-    }
+        Self: 'a;
 
     fn as_typed_slice_mut(&mut self) -> Option<&'a mut [T]>
     where
-        Self: 'a,
-    {
-        None
+        Self: 'a;
+}
+
+/// Trait for providing a rust typed pointer to an r type.
+/// R data are hidden behind [`SEXP`] opaque pointers with an identifier
+/// for the underlying r storage type. Thus this trait is only meaningful for
+/// [`SEXP`]-wrappers.
+trait SexpAs {
+    /// Identifier for the R object modes / types represented as a value.
```

Comment:
```suggestion
    /// Marker for the R object modes / types represented as a value.
```

I tend to expect an identifier is a string, although this is of course not always true

Reply:
I'm not married to this idea. You can resolve if you don't like it.